langfuse/_client/client.py

```diff
@@ -350,6 +360,108 @@ def start_span(
+                current-service propagation. Defaults to False.
+
+        Returns:
+            Context manager that sets values on all spans created within its scope.
+
+        Warning:
+            When as_baggage=True, the values will be included in HTTP headers of any
+            outbound requests made within this context. Only use this for non-sensitive
+            identifiers that are safe to transmit across service boundaries.
+
+        Example:
+           ```python
+            # Local context only (default)
+            with langfuse.with_attributes(session_id="session_123"):
+                with langfuse.start_as_current_span(name="process-request") as span:
+                    # This span and all its children will have session_id="session_123"
+                    child_span = langfuse.start_span(name="child-operation")
+
+            # Cross-service propagation (use with caution)
+            with langfuse.with_attributes(session_id="session_123", as_baggage=True):
+                # session_id will be propagated to external service calls
+                response = requests.get("https://api.example.com/data")
+            ```
+        """
+        current_context = otel_context_api.get_current()
+        current_span = otel_trace_api.get_current_span()
+
+        # Process session_id
+        if session_id is not None:
+            current_context = otel_context_api.set_value(
+                LANGFUSE_CTX_SESSION_ID, session_id, current_context
+            )
+            if current_span is not None and current_span.is_recording():
+                current_span.set_attribute("session.id", session_id)
+            if as_baggage:
+                current_context = otel_baggage_api.set_baggage(
+                    "session.id", session_id, current_context
+                )
+
+        # Process user_id
+        if user_id is not None:
+            current_context = otel_context_api.set_value(
+                LANGFUSE_CTX_USER_ID, user_id, current_context
+            )
+            if current_span is not None and current_span.is_recording():
+                current_span.set_attribute("user.id", user_id)
+            if as_baggage:
+                current_context = otel_baggage_api.set_baggage(
+                    "user.id", user_id, current_context
+                )
+
+        # Process metadata
+        if metadata is not None:
+            # Truncate values with size > 200 to 200 characters and emit warning including the ky
+            for k, v in metadata.items():
+                if not isinstance(v, str):
+                    # Ignore unreachable mypy warning as this runtime guard should make sense either way
+                    warnings.warn(  # type: ignore[unreachable]
+                        f"Metadata values must be strings, got {type(v)} for key '{k}'"
+                    )
+                    del metadata[k]
+                if len(v) > 200:
```

Comment:
We ran into an error here in production where `v` is `None` and this length check failed.

```
  File "/app/documents_parser/telemetry/middleware.py", line 41, in dispatch
    with client.with_attributes(
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/opentelemetry/util/_decorator.py", line 61, in __enter__
    return next(self.gen)  # type: ignore
           ^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/langfuse/_client/client.py", line 440, in with_attributes
    if len(v) > 200:
       ^^^^^^
TypeError: object of type 'NoneType' has no len()
```

Reply:
It's a bug on our side, but I think the package should probably warn and fail gracefully.